gammapy/utils/scripts.py

```diff
@@ -72,3 +74,40 @@ def get_all_main_functions():
         out[name] = module.main
 
     return out
+
+
+def set_up_logging_from_args(args):
+    """Set up logging from command line arguments.
+
+    This is a helper function that should be called from
+    all Gammapy command line tools.
+    It executes the boilerplate that's involved in setting
+    up the root logger the way we like it.
+    """
+    level = args.loglevel
+    del args.loglevel
+    _configure_root_logger(level=level)
+
+
+def _configure_root_logger(level='info', format=None):
+    """Configure root log level and format.
+
+    This is a helper function that can be called form
+    """
+    log = logging.getLogger()  # Get root logger
+
+    # Set log level
+    # level = getattr(logging, level.upper())
+    numeric_level = getattr(logging, level.upper(), None)
+    if not isinstance(numeric_level, int):
+        raise ValueError('Invalid log level: {}'.format(level))
+    log.setLevel(level=numeric_level)
+
+    # Format log handler
+    if not format:
+        # format = '%(asctime)s %(name)-12s %(levelname)-8s %(message)s'
+        format = '%(levelname)-8s %(message)s [%(name)s]'
+    formatter = logging.Formatter(format)
+    log.handlers[0].setFormatter(formatter)
```

Comment:
I'm getting this error from `gammapy-reflected-regions`:
```
$ gammapy-reflected-regions 
Traceback (most recent call last):
  File "/Users/deil/Library/Python/3.4/bin/gammapy-reflected-regions", line 9, in <module>
    load_entry_point('gammapy==0.3.dev1428', 'console_scripts', 'gammapy-reflected-regions')()
  File "/Users/deil/Library/Python/3.4/lib/python/site-packages/gammapy-0.3.dev1428-py3.4-macosx-10.10-x86_64.egg/gammapy/scripts/reflected_regions.py", line 36, in main
    set_up_logging_from_args(args)
  File "/Users/deil/Library/Python/3.4/lib/python/site-packages/gammapy-0.3.dev1428-py3.4-macosx-10.10-x86_64.egg/gammapy/utils/scripts.py", line 89, in set_up_logging_from_args
    _configure_root_logger(level=level)
  File "/Users/deil/Library/Python/3.4/lib/python/site-packages/gammapy-0.3.dev1428-py3.4-macosx-10.10-x86_64.egg/gammapy/utils/scripts.py", line 111, in _configure_root_logger
    log.handlers[0].setFormatter(formatter)
IndexError: list index out of range
```

From other command line scripts like e.g. `gammapy-find-obs` it works fine, no error.
I don't see any difference between the two.

@embray – I guess that's not a robust way to format the root logger? Any advice what to do instead?